actors/builtin/multisig/multisig_actor.go

```diff
@@ -476,33 +510,40 @@ func executeTransactionIfApproved(rt vmr.Runtime, st State, txnID TxnID, txn *Tr
 }
 
 type AddressResolveFunc func(address addr.Address) (resolved addr.Address, found bool)
+type SendFunc func(toAddr addr.Address, methodNum abi.MethodNum, params vmr.CBORMarshaler, value abi.TokenAmount) (vmr.SendReturn, exitcode.ExitCode)
 
-func isAddressEqual(resolveFunc AddressResolveFunc, addr1, addr2 addr.Address) bool {
-	return resolve(resolveFunc, addr1) == resolve(resolveFunc, addr2)
-}
-
-func isSigner(resolveFunc AddressResolveFunc, st *State, address addr.Address) bool {
-	candidateResolved := resolve(resolveFunc, address)
-
-	for _, ap := range st.Signers {
-		signerResolved := resolve(resolveFunc, ap)
-		if signerResolved == candidateResolved {
+func isSigner(address addr.Address, signers []addr.Address) bool {
+	AssertMsg(address.Protocol() == addr.ID, "address %v passed to isSigner must be a resolved address", address)
+	// signer addresses have already been resolved
+	for _, signer := range signers {
+		if signer == address {
 			return true
 		}
 	}
 
 	return false
 }
 
-func resolve(resolveFunc AddressResolveFunc, address addr.Address) addr.Address {
-	resolved := address
-	if resolved.Protocol() != addr.ID {
-		idAddr, found := resolveFunc(resolved)
-		if found {
-			resolved = idAddr
-		}
+func resolveToIDAddr(address addr.Address, resolveFunc AddressResolveFunc, sendFunc SendFunc) (addr.Address, error) {
```

Comment:
Move this to `builtin/shared.go`. Also it's ok for it to just take Runtime as a parameter, since it' always called at the top level and obviously needs it.

Reply:
This is done.